.pre-commit-config.yaml

```diff
@@ -1,7 +1,27 @@
 default_language_version:
     python: python3.9
 
+default_stages: [pre-commit, pre-push, manual]
+
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
```

Comment:
> not for Python code.

The editorconfig file is compatible with `black` style. As commented in https://github.com/sgl-project/sglang/pull/1926#issuecomment-2459066525, `black` is not enabled in our CI.

The `.editorconfig` file only defines a convention for the editor (e.g. vscode). It does not change the code files. The changes for Python code are made by `pre-commit` hook `trailing-whitespace` and `end-of-file-fixer`.